File: dart/advance/09.异步-05-async-await.dart
import 'dart:async';

void main(List<String> args) async {
  String _data = '0';
  Future<String> getData(int n) async {
    for (var i = 0; i < n; i++) {
      _data = '123';
    }
    return _data;
  }

  print('start');

  getData(100).then((value) => {print(value)});

  print('end');
}

File: dart/advance/09.异步-05-async-await.js

var _data='';
const getData = async (n) => {
  for (var i = 0; i < n; i++) {
    _data = '123';
  }
  console.log('_data: ', _data);
  return _data;
}


const  fn = async () =>{
  console.log('start')
  getData(100).then(el=>console.log(el))
  console.log('end')
}
fn()